009-ui-and-admin-server-security.md

```diff
@@ -0,0 +1,84 @@
+The login would not go through the Admin Server which hides the process from the external network but it may be desirable to add a `login` endpoint to the Admin Server which would allow session based applications to use the Admin Server in a natural way.
+
+The Strimzi Admin Server API could be called directly by the user creating an auth header on the request.
+Like the UI flow, the credentials would be retrieved from the auth header and placed into the Kafka client properties as SCRAM credentials.
+
+## OAuth 2
+
+![Kafka Client using Oauth2](./images/oauth-client.png)
+
+In the OAuth 2 scenario, Strimzi currently requires the user to configure the Kafka client with a clientId and secret.
+The client will then obtain an access token from the identity provider which is KeyCloak and that token flows to the broker.
+In the broker, calls to KeyCloak will be made based on the type of token that was requested by the client and passed to the broker.
+
+![Strimzi UI using Oauth2](./images/oauth-ui.png)
+
+In the UI model, the user will be presented with a login dialogue box when the UI is loaded into the browser and requires the user to enter a userId and password.
+The userId needs to be pre-defined to KeyCloak through the KeyCloak user interface.
+When the user enters their userId and password, the UI will contact KeyCloak to authorise the user and obtain an access token.
+This process will be controlled by `passport.js`.
+KeyCloak will either return an access token or reject the request.
+If an access token is returned, it will be stored in the session data for the session.
+When the UI requires data from the Admin Server, the token is recovered from the session data and placed as an auth header on the request.
+The Admin Server receives the request and configures the Kafka client in preparation for calling the Kafka broker.
+The token is retrieved from the auth header and is added to the Kafka client JAAS configuration properties before executing the call to the Kafka broker.
+The Kafka broker, which is running a custom SASL plugin, retrieves the token and sets up the Kafka Principal for the user.
+The request then passes through the KeyCloak Authorizer for authorization.
+
+In order to call the Admin Server directly, the user would need to obtain a token from KeyCloak manually and then store that token in an auth header on the call to the Admin Server.
+The behaviour when the AdminServer received the request would be identical to the flow for the UI calling the Admin Server with a token.
+
+## Mutual TLS
+
+![Admin server using Mutual TLS](./images/mtls-admin.png)
+
+With Mutual TLS, authentication takes place in the TLS handshake between the external web client and the Admin Server.
+The server side can only detect a userId for the client so credentials cannot be propagated to the Kafka Broker that woud allow the broker to authenticate and authorize the request.
```

Comment:
```suggestion
The server side can only detect a userId for the client so credentials cannot be propagated to the Kafka Broker that would allow the broker to authenticate and authorize the request.
```